doc/src/modules/combinatorics/pc_groups.rst

```diff
@@ -1,12 +1,295 @@
+the following form:
+
+* `v = {x_{i+1}}^{a_j}x_i`
+* `v = {x_{i+1}}^{a_j}{x_i}^{-1}`
+* `v = {x_i}^{a_j}`
+
+`a_j \notin \{0, \ldots relative\_order[j]-1\}`.
+
+>>> from sympy.combinatorics.named_groups import SymmetricGroup
+>>> from sympy.combinatorics.free_groups import free_group
+>>> G = SymmetricGroup(4)
+>>> PcGroup = G.polycyclic_group()
+>>> collector = PcGroup.collector
+>>> F, x1, x2 = free_group("x1, x2")
+>>> word = x2**2*x1**7
+>>> collector.minimal_uncollected_subword(word)
+((x2, 2),)
+
+
+Computation of Subword Index
+````````````````````````````
+
+For a given word and it's subword, subword_index computes the
+starting and ending index of the subword in the word.
+
+>>> from sympy.combinatorics.named_groups import SymmetricGroup
+>>> from sympy.combinatorics.free_groups import free_group
+>>> G = SymmetricGroup(4)
+>>> PcGroup = G.polycyclic_group()
+>>> collector = PcGroup.collector
+>>> F, x1, x2 = free_group("x1, x2")
+>>> word = x2**2*x1**7
+>>> w = x2**2*x1
+>>> collector.subword_index(word, w)
+(0, 3)
+>>> w = x1**7
+>>> collector.subword_index(word, w)
+(2, 9)
+
+
+Computation of Collected Word
+`````````````````````````````
+
+A word `W` is called collected, if `W = {x_{i_1}}^{a_1} \ldots {x_{i_r}}^{a_r}`
+with `i_1 < i_2< \ldots < i_r` and `a_j` is in `\{1 \ldots s_{j-1}\}`, where `s_j`
+represents the respective relative order.
+
+>>> from sympy.combinatorics.named_groups import SymmetricGroup
+>>> from sympy.combinatorics.perm_groups import PermutationGroup
+>>> from sympy.combinatorics.free_groups import free_group
+>>> G = SymmetricGroup(4)
+>>> PcGroup = G.polycyclic_group()
+>>> collector = PcGroup.collector
+>>> F, x0, x1, x2, x3 = free_group("x0, x1, x2, x3")
+>>> word = x3*x2*x1*x0
+>>> collected_word = collector.collected_word(word)
+>>> free_to_perm = {}
+>>> free_group = collector.free_group
+>>> for sym, gen in zip(free_group.symbols, collector.pcgs):
+...     free_to_perm[sym] = gen
+>>> G1 = PermutationGroup()
+>>> for w in word:
+...     sym = w[0]
+...     perm = free_to_perm[sym]
+...     G1 = PermutationGroup([perm] + G1.generators)
+>>> G2 = PermutationGroup()
+>>> for w in collected_word:
+...     sym = w[0]
+...     perm = free_to_perm[sym]
+...     G2 = PermutationGroup([perm] + G2.generators)
+>>> G1 == G2
+True
+
+
+Computation of Polycyclic Presentation
+--------------------------------------
+
+The computation of presentation starts from the bottom of the pcgs and polycyclic series.
+Storing all the previous generators from pcgs and then taking the last generator
+as the generator which acts as a conjugator and conjugates all the previous
+generators in the list.
+
+To get a clear picture, start with an example of SymmetricGroup(4). For S(4) there are 4
+generators in pcgs say [x0, x1, x2, x3] and the relative_order vector is [2, 3, 2, 2].
+Starting from bottom of this sequence the presentation is computed in order as below.
+
+>>> x3^2	   	---|
```

Comment:
It looks like there are tabs. They should be removed. In addition, the comments should start with `#`.